utt/include/client.hpp

```diff
@@ -0,0 +1,52 @@
+#pragma once
+#include "commitment.hpp"
+#include "globalParams.hpp"
+#include "common.hpp"
+#include "types.hpp"
+#include <memory>
+namespace libutt {
+class AddrSK;
+class RandSigPK;
+class RegAuthPK;
+}  // namespace libutt
+namespace libutt::api {
+class Coin;
+namespace operations {
+class Burn;
+class Transaction;
+}  // namespace operations
+class Client {
+ public:
+  Client(const std::string& pid,
+         const std::string& bpk,
+         const std::string& rpk,
+         const std::string& csk,
+         const std::string& mpk);
+  Commitment generateInputRCM();
+  void setPRFKey(const types::CurvePoint& s2);
+  const std::string& getPid() const;
+  types::CurvePoint getPRFSecretKey() const;
+  types::CurvePoint getPidHash() const;
+
+  // This method has to be called only after setPRFKey with s2
```

Comment:
Instead of relying on this comment and hope that the caller won't do mistakes can we combine setPRFKey and setRCMSig to provide a single method that does the right thing?

Reply:
fixed